testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/EEConcurrencyExecutorShutdownTestCase.java

```diff
@@ -0,0 +1,230 @@
+ *
+ * @author Daniel Cihak
+ */
+public class EEConcurrencyExecutorShutdownTestCase {
+
+    private static final String ARCHIVE_FILE_NAME = "test.war";
+    private static final PathAddress ROOT_DEPLOYMENT_ADDRESS = PathAddress.pathAddress(DEPLOYMENT, ARCHIVE_FILE_NAME);
+    private static final PathAddress MAIN_SERVER_GROUP_ADDRESS = PathAddress.pathAddress(SERVER_GROUP, "main-server-group");
+    private static final PathAddress MAIN_SERVER_GROUP_DEPLOYMENT_ADDRESS = MAIN_SERVER_GROUP_ADDRESS.append(DEPLOYMENT, ARCHIVE_FILE_NAME);
+    private static final String FIRST_SERVER_NAME = "main-one";
+
+    private static DomainTestSupport testSupport;
+    private static DomainLifecycleUtil domainMasterLifecycleUtil;
+    private static File tmpDir;
+
+    private static DomainClient masterClient;
+
+    @BeforeClass
+    public static void setupDomain() {
+        testSupport = DomainTestSuite.createSupport(EEConcurrencyExecutorShutdownTestCase.class.getSimpleName());
+        domainMasterLifecycleUtil = testSupport.getDomainMasterLifecycleUtil();
+        masterClient = domainMasterLifecycleUtil.getDomainClient();
+    }
+
+    @Before
+    public void createDeployment() throws Exception {
+        WebArchive testDeployment = ShrinkWrap.create(WebArchive.class, ARCHIVE_FILE_NAME);
+        testDeployment.addClasses(TaskSchedulerServletContextListener.class);
+        tmpDir = new File("target/deployments/" + this.getClass().getSimpleName());
+        new File(tmpDir, "archives").mkdirs();
+        testDeployment.as(ZipExporter.class).exportTo(new File(tmpDir, "archives/" + ARCHIVE_FILE_NAME), true);
+
+        final WildFlyManagedConfiguration config = createConfiguration("domain.xml", "host.xml", getClass().getSimpleName());
+        config.setAdminOnly(true);
+        config.setReadOnlyDomain(true);
+        config.setReadOnlyHost(true);
+
+        // Trigger the servers to fail on boot if there are runtime errors
+        String hostProps = config.getHostCommandLineProperties();
+        hostProps = hostProps == null ? "" : hostProps;
+        config.setHostCommandLineProperties(hostProps + "\n-Djboss.unsupported.fail-boot-on-runtime-failure=true");
+
+        this.addTestResources();
+    }
+
+    @After
+    public void removeTestResources() throws IOException, MgmtOperationException {
+        try {
+            // /profile=default/subsystem=ee/context-service=default:remove()
+            ModelNode removeContextService = createOpNode("profile=default/subsystem=ee/context-service=default", REMOVE);
+            executeForResult(removeContextService, masterClient);
+            // /profile=default/subsystem=ee/managed-executor-service=default:remove()
+            ModelNode removeManagedExecutorService = createOpNode("profile=default/subsystem=ee/managed-executor-service=default", REMOVE);
+            executeForResult(removeManagedExecutorService, masterClient);
+        } finally {
+            testSupport.stop();
```

Comment:
I think there are mixed responsibilities in this method. `removeTestResources()` should be used to remove the resources you have added, but it is stopping the domain. I suggest to use `testSupport.stop();` in `tearDownDomain()` instead.